packages/design-system/src/components/dropDown/index.js

```diff
@@ -102,7 +126,8 @@ export const DropDown = forwardRef(
     const handleDismissMenu = useCallback(() => {
       isOpen.set(false);
       internalRef.current.focus();
-    }, [isOpen, internalRef]);
+      resetPlacement();
+    }, [isOpen, resetPlacement]);
```

Comment:
I don't think we'll need this `recetPlacement()` logic if we just do `placement={dynamicPlacement || placement}` and make sure to set `dynamicPlacement` to `null` when it doesn't meet the particular conditions